pkg/utils/kubernetes/reconciler.go

```diff
@@ -0,0 +1,120 @@
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+	"github.com/RHsyseng/operator-utils/pkg/resource"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strings"
+)
+
+type Finalizer interface {
+	GetName() string
+	OnFinalize(owner resource.KubernetesResource, service PlatformService) error
+}
+
+type ExtendedReconciler struct {
+	Service    PlatformService
+	Reconciler reconcile.Reconciler
+	Resource   resource.KubernetesResource
```

Comment:
Not sure about `Resource` here. Looks like you can only finalize the operator's primary CRD type. Why not any arbitrary object?

Reply:
If an owned object also has a finalizer that means the object hierarchy is getting too complex. I don't think that case is very common. If an operator defines multiple CRDs, it is mostly implemented as multiple reconcilers which is still fine. What do you think?

Reply:
Right, I meant an owned object. As mentioned before, I think the full reconcile of the KIE cloud operator means there is no need for a finalizer there, but if we were thinking of it as a use case, the finalizer would actually be on the business central deployment and not the CR.

Reply:
That implies that when receiving a reconcile request we get all the object types that have at least one registered finalizer to check if such objects are being deleted because the reconcile request doesn't specify the type, just the namespacedName.